File: fizzbuzz/java/src/main/java/fizzbuzz/FizzBuzz.java
package fizzbuzz;

class FizzBuzz {

    static String say(int number) {

        if (number < 1) {
            throw new IllegalArgumentException();
        }

        if (number % 15 == 0) {
            return "fizzbuzz";
        } else if (number % 3 == 0) {
            return "fizz";
        } else if (number % 5 == 0) {
            return "buzz";
        }

        return String.valueOf(number);
    }
}

File: fizzbuzz/kotlin/src/main/kotlin/fizzbuzz/FizzBuzz.kt
package fizzbuzz

fun fizzBuzz(number: Int): String = when {
    number < 1 -> throw IllegalArgumentException()
    number % 15 == 0 -> "fizz-buzz"
    number % 3 == 0 -> "fizz"
    number % 5 == 0 -> "buzz"
    else -> "$number"
}
